File: src/main/kotlin/com/itmo/mpa/entity/Disease.java
package com.itmo.mpa.entity;


import org.jetbrains.annotations.NotNull;

import javax.persistence.*;

@Entity
@Table(name = "diseases")
public class Disease {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private long id;

    @Column(name = "Name", nullable = false)
    private String name;

    public Disease() {
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @NotNull
    public String getName() {
        return name;
    }

    public void setName(@NotNull String name) {
        this.name = name;
    }
}

File: src/main/kotlin/com/itmo/mpa/entity/Disease.kt
package com.itmo.mpa.entity

import com.itmo.mpa.entity.medicine.Medicine
import javax.persistence.*

@Entity
@Table(name = "disease")
class Disease : LongIdEntity() {

    @Column(name = "name", nullable = false)
    lateinit var name: String

    @ManyToMany(fetch = FetchType.EAGER, cascade = [CascadeType.PERSIST, CascadeType.MERGE])
    @JoinTable(name = "medicine_for_diseases",
            joinColumns = [JoinColumn(name = "disease_id", nullable = false)],
            inverseJoinColumns = [JoinColumn(name = "medicine_id", nullable = false)])
    lateinit var medicines: Set<Medicine>
}
